File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {

    private static final double BASE_PRODUCTION_RATE = 221.0;

    public double productionRatePerHour(int speed) {
        double baseProduction = speed * BASE_PRODUCTION_RATE;

        if (speed <= 4) {
            return baseProduction;
        }
        else if (speed <= 8) {
            return baseProduction * 0.9;
        }

        return speed == 9
          ? baseProduction * 0.8
          : baseProduction * 0.77;
    }

    public int workingItemsPerMinute(int speed) {
        return (int)Math.floor(productionRatePerHour(speed)) / 60;
    }
}

File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
using System;

static class AssemblyLine
{
    private const int BaseProductionRate = 221;
    private const int MinutesPerHour     = 60;

    public static double ProductionRatePerHour(int speed)
        => speed * BaseProductionRate * GetSuccessRate(speed);

    public static int WorkingItemsPerMinute(int speed)
        => (int) Math.Floor(ProductionRatePerHour(speed) / MinutesPerHour);

    private static double GetSuccessRate(int speed)
        => speed switch
        {
            <  5 => 1.00,
            <  9 => 0.90,
            < 10 => 0.80,
            _    => 0.77
        };
}
